### PR TITLE
Feature sequelize

### PR DESCRIPTION
**Pull request by selva-bct**

### CODE REVIEW

File: user-module/constant/constant.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ export const defaultStatusCode = {
   NOT_AUTHORIZED: 401,
   INTERNAL_SERVER_ERROR: 500,
   SUCCESS: 200,
+  ERROR: 204,
```
Comment (selva-bct) [inline user-module/constant/constant.js]: change this to 500

File: user-module/service/cognito-service.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        ClientId: cognito.appClientId,
+        AuthParameters: {
+          USERNAME: username,
+          PASSWORD: password
+        }
+      }
+      return await promisify(this.cognitoClient.adminInitiateAuth.bind(this.cognitoClient, params))()
+    } catch (error) {
+      logger.error('Error while login user')
+      this.updateFailAttempts(username)
```
Comment (selva-bct) [inline user-module/service/cognito-service.js]: this should happen in user controller

File: user-module/controller/user-controller.js [hunk 3]
```diff
@@ -88,6 +80,7 @@ class UserController {
       logger.info('Admin setting user email verified as true')
       const param = { UserPoolId: cognito.userPoolId, Username: name, UserAttributes: [{ Name: 'email_verified', Value: 'true' }] }
       await promisify(this.cognitoClient.adminUpdateUserAttributes.bind(this.cognitoClient, param))()
+
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: move the `adminCreateUser`, `adminSetUserPassword`, `adminUpdateUserAttributes` to cognito service

File: user-module/controller/user-controller.js [hunk 4]
```diff
@@ -123,20 +116,28 @@ class UserController {
...
-      const { appClientId } = cognito
-      const params = {
-        ClientId: appClientId,
-        Username: username
-      }
-      const data = await promisify(this.cognitoClient.forgotPassword.bind(this.cognitoClient, params))()
+      cognotiService.forgotPassword(username)
       logger.info(resetPasswordRequest.SUCCESS)
-      return responseService.onSuccess(res, resetPasswordRequest.SUCCESS, data)
+      return responseService.onSuccess(res, resetPasswordRequest.SUCCESS, defaultStatusCode.SUCCESS)
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: why is this status code been passed to response service

File: user-module/controller/user-controller.js [hunk 5]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async updateUser (req, res) {
+    try {
+      logger.info('Updating user')
+      const user = req.body
+      const userInfo = await userService.getUser(user.username)
+      if (userInfo !== null) {
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: update this code to `if (!userInfo)`

File: user-module/controller/user-controller.js [hunk 6]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async getUserList (req, res) {
+    try {
+      logger.info('Getting user list')
+      const data = await userService.getUserList()
+      logger.info(defaultMessage.SUCCESS)
+      responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: remove the default success code

File: user-module/controller/user-controller.js [hunk 7]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+  async updateUser (req, res) {
+    try {
+      logger.info('Updating user')
+      const user = req.body
+      const userInfo = await userService.getUser(user.username)
+      if (userInfo !== null) {
+        const data = await userService.updateUser(user)
+        logger.info(defaultMessage.SUCCESS)
+        responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+      } else {
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: wrong error response

File: user-module/controller/user-controller.js [hunk 8]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+      const data = await userService.getUser(username)
+      logger.info(defaultMessage.SUCCESS)
+      responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+    } catch (error) {
+      logger.error(error, defaultMessage.ERROR)
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async updateUser (req, res) {
```
Comment (selva-bct) [inline user-module/controller/user-controller.js:157]: please check this serviice with positive and negative cases.

File: user-module/controller/user-controller.js [hunk 9]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+
+  async getUser (req, res) {
+    try {
+      logger.info('Getting user by Email')
+      const { username } = req.body
+      const data = await userService.getUser(username)
+      logger.info(defaultMessage.SUCCESS)
+      responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+    } catch (error) {
+      logger.error(error, defaultMessage.ERROR)
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: for all error handling please send the error message from the error object

File: user-module/controller/user-controller.js [hunk 10]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async addRole (req, res) {
+    try {
+      logger.info('Adding new role')
+      if (req.body.role !== null && req.body.role !== '') {
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: use object destruction to get role form the req body.
null check can be done in the following format
`if (!role)` Please use this syntax to null check else where

File: user-module/controller/user-controller.js [hunk 11]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+        const data = await userService.addRole(req.body)
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: dont pass in req.body in to another method. assigni t o another variable and then pass the new variable

File: user-module/controller/user-controller.js [hunk 12]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+        logger.info(defaultMessage.SUCCESS)
+        responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+      } else {
+        logger.error(defaultMessage.ERROR)
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: wrong error response

File: user-module/controller/user-controller.js [hunk 13]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+      const data = await userService.getUserList()
+      logger.info(defaultMessage.SUCCESS)
+      responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+    } catch (error) {
+      logger.error(error, defaultMessage.ERROR)
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async addRole (req, res) {
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: before adding a role check if the role exists in db

File: user-module/controller/user-controller.js [hunk 14]
```diff
@@ -171,5 +172,129 @@ class UserController {
...
+      logger.error(error, defaultMessage.ERROR)
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async updateRole (req, res) {
+    try {
+      logger.info('Updating role')
+      const role = req.body
+      const roleInfo = await userService.getRole(role.roleId)
```
Comment (selva-bct) [inline user-module/controller/user-controller.js]: pick the role id in params

File: user-module/constant/constant.js (hunk 1 above)
Comment (Eswar-BCT) [reply user-module/constant/constant.js]: done

File: user-module/controller/user-controller.js (hunk 3 above)
Comment (Eswar-BCT) [reply user-module/controller/user-controller.js]: done

**Comment by Eswar-BCT** [comment]
verified changes and merging to master

**Pull request merged**